source/adios2/core/IO.cpp

```diff
@@ -467,24 +514,47 @@ Engine &IO::Open(const std::string &name, const Mode mode, MPI_Comm mpiComm)
                        engineTypeLC.begin(), ::tolower);
     }
 
+    /* Second step in handling virtual engines */
     /* BPFile for read needs to use BP4 or BP3 depending on the file's version
      */
-    if ((engineTypeLC == "bpfile" || engineTypeLC == "bp" || isDefaultEngine))
+    if ((engineTypeLC == "file" || engineTypeLC == "bpfile" ||
+         engineTypeLC == "bp" || isDefaultEngine))
     {
-        if (mode == Mode::Read)
+        if (helper::EndsWith(name, ".h5", false))
+        {
+            engineTypeLC = "hdf5";
+        }
+        else if (mode == Mode::Read)
         {
             if (adios2sys::SystemTools::FileIsDirectory(name))
             {
                 engineTypeLC = "bp4";
             }
             else
             {
-                engineTypeLC = "bp3";
+                if (helper::EndsWith(name, ".bp", false))
+                {
+                    engineTypeLC = "bp3";
```

Comment:
Should also probably check if it's a directory or not so you can have `foo.bp` as a file use bp3 and as a directory use bp4.